hydra-explorer/src/Hydra/Explorer/ObservationApi.hs

```diff
@@ -0,0 +1,45 @@
+-- | Types for the observer-side REST API.
+module Hydra.Explorer.ObservationApi where
+
+import Hydra.Prelude
+
+-- XXX: Depending on hydra-tx:testlib feels wrong here.
+import Test.Hydra.Tx.Gen ()
+
+import Hydra.Cardano.Api (BlockNo, ChainPoint, NetworkId (..), NetworkMagic (..), Tx, fromNetworkMagic)
+import Hydra.Chain (OnChainTx)
+import Servant (Capture, FromHttpApiData (..), JSON, Post, ReqBody, (:>))
+
+data Observation = Observation
+  { point :: ChainPoint
+  , blockNo :: BlockNo
+  , observedTx :: Maybe (OnChainTx Tx)
+  }
+  deriving stock (Eq, Show, Generic)
+  deriving anyclass (FromJSON)
+
+instance Arbitrary Observation where
+  arbitrary = Observation <$> arbitrary <*> arbitrary <*> arbitrary
+
+-- | New type wrapper for defining API instances over a 'NetworkId'.
+newtype NetworkParam = NetworkParam NetworkId
+
+instance FromHttpApiData NetworkParam where
+  parseUrlPiece = \case
+    "mainnet" -> Right $ NetworkParam Mainnet
+    t -> maybe (Left $ "failed to parse network magic: " <> show t) Right $ do
+      magic <- readMaybe $ toString t
+      pure $ NetworkParam $ fromNetworkMagic $ NetworkMagic magic
+
+-- | Version of a hydra head as reported by chain observers.
+newtype HydraVersion = HydraVersion Text
+  deriving stock (Eq, Show, Generic)
+  deriving newtype (FromJSON, ToJSON, Arbitrary, FromHttpApiData)
+
+-- TODO: test correspondence with openapi
```

Comment:
I'm not sure how to do this best. I do _not_ want to directly generate the openapi spec from the servant API, but some kind of equivalence check would be good to have?

Reply:
Created a follow-up issue https://github.com/cardano-scaling/hydra-explorer/issues/21

Reply:
This test is equivalent to generating servant -> openapi, no? The reverse is harder.